tests/state_test_util.go

```diff
@@ -325,17 +325,28 @@ func (t *stTransaction) At(i indexes, baseFee *big.Int) (*types.Transaction, err
 		value = v
 	}
 
-	return types.NewTx(&types.MixedTxn{
-		From:      t.From,
-		To:        t.To,
-		Nonce:     t.Nonce,
-		Value:     value,
-		Gas:       t.GasLimit[i.Gas],
-		GasPrice:  gasPrice,
-		GasFeeCap: t.MaxFeePerGas,
-		GasTipCap: t.MaxPriorityFeePerGas,
-		Input:     hex.MustDecodeHex(t.Data[i.Data]),
-	}), nil
+	if t.MaxFeePerGas != nil && t.MaxPriorityFeePerGas != nil {
+		return types.NewTx(&types.DynamicFeeTx{
+			From:      t.From,
+			To:        t.To,
+			Nonce:     t.Nonce,
+			Value:     value,
+			Gas:       t.GasLimit[i.Gas],
+			GasFeeCap: t.MaxFeePerGas,
+			GasTipCap: t.MaxPriorityFeePerGas,
+			Input:     hex.MustDecodeHex(t.Data[i.Data]),
+		}), nil
+	} else {
+		return types.NewTx(&types.LegacyTx{
+			From:     t.From,
+			To:       t.To,
+			Nonce:    t.Nonce,
+			Value:    value,
+			Gas:      t.GasLimit[i.Gas],
+			GasPrice: t.GasPrice,
+			Input:    hex.MustDecodeHex(t.Data[i.Data]),
+		}), nil
+	}
```

Comment:
Nitpick, you can even get rid off the else statement if you wish:

```suggestion
	} 
    
    return types.NewTx(&types.LegacyTx{
			From:     t.From,
			To:       t.To,
			Nonce:    t.Nonce,
			Value:    value,
			Gas:      t.GasLimit[i.Gas],
			GasPrice: t.GasPrice,
			Input:    hex.MustDecodeHex(t.Data[i.Data]),
		}), nil
```